source/include/shadow.h

```diff
@@ -265,362 +307,342 @@ typedef enum ShadowStatus
 
 /**
- * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update" or
- *        "$aws/things/<thingName>/shadow/name/<shadowName>/update".
+ * @ingroup shadow_constants
+ * @brief Compute the length of the longest shadow topic.
  *
  * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
  * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
  *
  * @return Length of the shadow topic in bytes.
  */
-#define SHADOW_TOPIC_LENGTH_UPDATE( thingNameLength, shadowNameLength ) \
-    SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_NULL_LENGTH, thingNameLength, shadowNameLength )
+#define SHADOW_TOPIC_LENGTH_MAX( thingNameLength, shadowNameLength ) \
+    SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_DOCUMENTS_LENGTH, thingNameLength, shadowNameLength )
 
 /**
- * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update/accepted" or
- *        "$aws/things/<thingName>/shadow/name/<shadowName>/update/accepted".
+ * @ingroup shadow_constants
+ * @brief Assemble constant shadow topic strings when Thing Name and Shadow Name are known at compile time.
  *
- * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
- * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
+ * When thingName and shadowName are known to be "myThing" and "myShadow" at compile time, invoke the macro like this:
  *
- * @return Length of the shadow topic in bytes.
+ *     SHADOW_TOPIC_STRING( SHADOW_OP_UPDATE_DELTA, "myThing", "myShadow" )
+ *
+ * When thingName and/or shadowName is only known at run time, do not use this macro. Use the
+ * Shadow_AssembleTopicString() function instead.
+ *
+ * @param[in] operation     Can be one of:
+ *                              - #SHADOW_OP_UPDATE
+ *                              - #SHADOW_OP_UPDATE_ACCEPTED
+ *                              - #SHADOW_OP_UPDATE_REJECTED
+ *                              - #SHADOW_OP_UPDATE_DOCUMENTS
+ *                              - #SHADOW_OP_UPDATE_DELTA
+ *                              - #SHADOW_OP_GET
+ *                              - #SHADOW_OP_GET_ACCEPTED
+ *                              - #SHADOW_OP_GET_REJECTED
+ *                              - #SHADOW_OP_DELETE
+ *                              - #SHADOW_OP_DELETE_ACCEPTED
+ *                              - #SHADOW_OP_DELETE_REJECTED
+ *
+ * @param[in] thingName     Thing Name.
+ * @param[in] shadowName    Shadow Name. Empty string for an unnamed ("Classic") shadow.
+ *
+ * @return Topic string.
  */
-#define SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( thingNameLength, shadowNameLength ) \
-    SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_ACCEPTED_LENGTH, thingNameLength, shadowNameLength )
+#define SHADOW_TOPIC_STRING( operation, thingName, shadowName )            \
```

Comment:
As discussed offline, backward compatibility for the existing MACRO will be good.

Reply:
Done.